### PR TITLE
Removing all instances of unittest

### PR DESCRIPTION
**Pull request by prady0t**
WIP, will be adding ruff rule here.

### CODE REVIEW

**Comment by prady0t** [comment]
I've added two rules, PT027 and PT009. These won't detect the usage of unittest, but is enough to prevent someone from writing tests in future as it prevents using unittest style assertions.

File: pyproject.toml [hunk 1]
```diff
@@ -196,6 +196,8 @@ extend-select = [
     "YTT",           # flake8-2020
     "TID252",        # relative-imports
     "S101",          # to identify use of assert statement
+    "PT027",         # remove unittest style assertion
+    "PT009",         # Use pytest.raises instead of unittest-style
```
Comment (Saransh-cpp) [inline pyproject.toml:200]: Can you instead try turning on the "PT" rule commented above and see what happens?

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
This looks amazing, thanks, @prady0t!

File: pyproject.toml (hunk 1 above)
Comment (prady0t) [reply pyproject.toml:200]: There are a lot of style failures. Should I try fixing those and keep the `PT` rule?

Comment (agriyakhetarpal) [reply pyproject.toml:200]: If there are too many, then a separate PR for this would be better.

Comment (Saransh-cpp) [reply pyproject.toml:200]: A separate PR sounds good then. Could you create one after this, @prady0t?

**Review by Saransh-cpp — APPROVED** [review]

**Pull request merged**